include/glow/Optimizer/GraphOptimizer/Pipeline/Pipeline.h

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_OPTIMIZER_GRAPHOPTIMIZER_GRAPHOPTIMIZER_PIPELINES_PIPELINES_H
+#define GLOW_OPTIMIZER_GRAPHOPTIMIZER_GRAPHOPTIMIZER_PIPELINES_PIPELINES_H
+
+#include "glow/Optimizer/GraphOptimizer/CompilationContext.h"
+#include "glow/Support/Support.h"
+
+#include <bitset>
+
+namespace glow {
+
+/// Define an enum to identify all FunctionPasses, used to declare inside a
+/// FunctionPassConfig for a Pipeline.
+enum class FunctionPassID {
+#define FUN_PASS(PASS_NAME) PASS_NAME,
+#include "glow/Optimizer/GraphOptimizer/FunctionPasses.def"
+};
+
+/// Specifies convergence mode for a FunctionPass.
+enum class ConvergenceMode {
+  OnePass,         // Run a single pass over the Function.
+  UntilFixedPoint, // Run the pass over the Function until a fixed point is
+                   // reached.
+};
+
+/// Specifies a configuration for running a FunctionPass when used in a
+/// FunctionPassPipeline.
+class FunctionPassConfig {
+public:
+  /// Specifies whether the pass requires DCE.
+  enum class DCERequiredMode {
```

Comment:
Maybe move it before `FunctionPassConfig`, right after `ConvergenceMode`? I just don't see any reason for one of them to be defined outside and another one inside `FunctionPassConfig`.

Reply:
I thought DCERequiredMode was more specific to FunctionPass, while ConvergenceMode might be used by future Module/IRFunction passes. But NBD, can move it out.